src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentRefactoringService.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    public interface IDocumentRefactoringService : IWorkspaceService
+    {
+        /// <summary>
+        /// Updates the contents of a document after some document info change, such as file name or path. May
+        /// prompt a user for confirmation on the changes.
```

Comment:
Hmm, this is a data API -- it really shouldn't be prompting. (I wish you luck when this is on a server...)

Reply:
If it needs use input, this seems like the best place for that information to be obtained. How does this directly vary from `CodeActionWithOptions` or similar?

Reply:
ping @jasonmalinowski